packages/workbox-precaching/src/utils/cleanRedirect.ts

```diff
@@ -20,9 +21,9 @@ export async function cleanRedirect(response: Response) {
 
   // Not all browsers support the Response.body stream, so fall back
   // to reading the entire body into memory as a blob.
-  const bodyPromise = 'body' in clonedResponse ?
+  const bodyPromise = browserSupportsStreams() ?
```

Comment:
I'm concerned that this isn't checking the same thing—it's at least possible to imagine a browser in which `new ReadableStream({start() {}});` executes properly, but in which `clonedResponse` doesn't contain a `body` property.

Can you change this to `'body' in clonedResponse && browserSupportsStreams()` just to be safe?

Reply:
What about having that check in the `browserSupportsStreams()` core function? Putting all the detection logic there would allow us to only use streams if they're fully supported for all our use cases.

Or do you think there may be situations where we're OK with partial support?

Reply:
I believe that these are all separate, potentially independently implemented features:

- Support for constructing a new `ReadableStream` from scratch.
- The `body` property of an existing `Response` can be read as a stream.
- Using a `ReadableStream` as the `BodyInit` when constructing a new `Response`.

CC: @wanderview who might be able to provide some more insight there, because I think that Firefox (at one time, at least) only supported a subset of those features, and I remember being bit by making assumptions in the past.

With this whole area, being explicit about checking for what we need (and only what we need) in order to enable a specific type of functionality seems the safest. We could end up regressing if we impose checks that are too stringent.

Reply:
At least for Firefox it was just that there was a separate flag to enable the constructor-vs-fetch-integrations.  So this:

* Support for constructing a new ReadableStream from scratch.

Might be true, but these might be false:

* The body property of an existing Response can be read as a stream.
* Using a ReadableStream as the BodyInit when constructing a new Response.

But these later two would either both be false or true at the same time.

At this point I believe FF has shipped all three of these and no shipping FF had any of the flags flipped with other configurations.

Hope that helps.

Reply:
> With this whole area, being explicit about checking for what we need (and only what we need) in order to enable a specific type of functionality seems the safest.

But that's exactly what we did and what caused this bug in Edge. We assumed that the presence of `response.body` meant we could make a new `Response` from it, and that's what was failing in Edge.

Given that all browsers other than Edge currently support all of these stream-related features, and given that we already have a bug (this one) from making assumptions about partial stream support, I think it makes sense for us to have a robust `areStreamsFullySupported()` function that does a full check, and then we only use stream-related features if they all pass.

> We could end up regressing if we impose checks that are too stringent.

I can see how it might be hypothetically possible for a browser to support a subset of these features in a way that doing what I describe might not be the 100% most efficient way for some partial-support use case, but from looking at our code I don't think that is the case for us (the uses in the PR are the only streams-related cases AFAIK).

How about doing what I suggest above and then changing to a more granular check if we discover a browser with partial support that would benefit from it?